### ISSUE TITLE
Training loop evaluates on the test set

### ISSUE THREAD

**Issue by julianmack** [punyajoy/HateXplain#2]
I've been looking at your paper and repo - I think it's a very important contribution to HS detection and explanation (congrats on the AAAI acceptance :tada: !). 

However I've noticed that your training script (+ example notebooks) evaluate models on the test set every epoch. IMO - to prevent researchers overfitting to the test set, evaluation should only be regularly run on the train & dev sets (perhaps with a separate script to run test set eval).

**Comment by punyajoy** [comment]
Thanks for the suggestion. Although I agree that a separate script will be better but if you check our code, we only evaluate the model based on the dev set performance.  
As for preventing researchers from overfitting on our dataset,  I think we have less control over that as the test data has ground truth labels also.
We will try to incorporate the changes in subsequent releases
